app/Http/Controllers/Backend/SemesterController.php

```diff
@@ -0,0 +1,112 @@
+<?php
+
+namespace App\Http\Controllers\Backend;
+
+use App\Http\Controllers\Controller;
+use Illuminate\Http\Request;
+use App\Domains\Semester\Models\Semester;
+
+class SemesterController extends Controller
+{
+    /**
+     * Display a listing of the semesters.
+     *
+     * @return \Illuminate\Http\Response
+     */
+    public function index()
+    {
+        $semesters = Semester::all();
+        return view('backend.semesters.index', compact('semesters'));
+    }
+
+    /**
+     * Show the form for creating a new semester.
+     *
+     * @return \Illuminate\Http\Response
+     */
+    public function create()
+    {
+        return view('backend.semesters.create');
+    }
+
+    /**
+     * Store a newly created semester in storage.
+     *
+     * @param  \Illuminate\Http\Request  $request
+     * @return \Illuminate\Http\Response
+     */
+    public function store(Request $request)
+    {
+        $validatedData = $request->validate([
+            'title' => 'required|string|max:255',
+            'version' => 'required|integer',
+            'academic_program' => 'required|in:Undergraduate,Postgraduate',
+            'description' => 'nullable|string',
+            'url' => 'nullable|string',
+        ]);
+        try {
+            Semester::create($validatedData);
+            return redirect()->route('dashboard.semesters.index')->with('success', 'Semester created successfully.');
+        } catch (\Exception $e) {
+            return abort(500);
+        }
+    }
+
+    
+    /**
+     * Show the form for editing the specified semester.
+     *
+     * @param  \App\Domains\Semester\Models\Semester  $semester
+     * @return \Illuminate\Http\Response
+     */
+    public function edit(Semester $semester)
+    {
+        return view('backend.semesters.edit', compact('semester'));
+    }
+
+    /**
+     * Update the specified semester in storage.
+     *
+     * @param  \Illuminate\Http\Request  $request
+     * @param  \App\Domains\Semester\Models\Semester  $semester
+     * @return \Illuminate\Http\Response
+     */
+    public function update(Request $request, Semester $semester)
+    {
+        $validatedData = $request->validate([
+            'title' => 'required|string|max:255',
+            'version' => 'required|integer',
+            'academic_program' => 'required|in:Undergraduate,Postgraduate',
+            'description' => 'nullable|string',
+            'url' => 'nullable|string',
```

Comment:
Check the unique 

 'url' => ['nullable','string'. Rule::unique('semester')->ignore($semester->id)],